__1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):<br>
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],<br>
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].<br>
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.__

In [1]:
import numpy as np
import math
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

print(f"M(x) = {zp.mean()}")

print(f"M(y) = {ks.mean()}")

print(f"M(xy) = {(zp * ks).mean()}")

cov = (zp * ks).mean() - zp.mean() * ks.mean()
print(f"Коввариация - cov(xy) = M(xy) - M(x)*M(y) = {cov}")

M(x) = 101.4
M(y) = 709.9
M(xy) = 81141.7
Коввариация - cov(xy) = M(xy) - M(x)*M(y) = 9157.839999999997


In [2]:
np.cov(zp, ks, ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [3]:
print(f"коэффициент корреляции Пирсона = {cov / (zp.std(ddof=0) * ks.std(ddof=0)): .4f}")

коэффициент корреляции Пирсона =  0.8875


Вывод: $r_{xy} = 0.8875; r_{xy}->1, r_{xy}>\frac{1}{2} =>$ <br>
Значения связаны линейно $=>$<br>
Между zp и ks сильная прямая корреляционная связь. При возрастании xp - возрастает ks и при убывании zp - убывает ks.

__2. Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:<br>
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.<br>
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.__

$p=0.95 => \alpha = 1-p = 0.05$<br>
$n=10$<br>
$[Xв - K \cdot Ст.ош.ср; Xв - K \cdot Ст.ош.ср]$<br>
1) Неизвестна $\sigma$ => небыло анализа ГС.<br>
2) Из пункта 1) => критерий оценки K это критерий <font color = red>Стьюдента t</font><br>
$K=n-1 => K=10-1=9$<br>
3) Из 1) => $Ст.ош.ср. = \dfrac{s}{\sqrt{n}}$, где<br>
$s = \sqrt{D(x)}$, где<br>
$\sqrt{D(x)}$ - несмещенная оценка дисперсии, выполненная <font color = red>по выборке</font><br>
4) Испытания выполнены с одинаковой точностью => ГС распределена нормально<br>
5) Из 4) =>  от $\alpha$ переходим к $\frac{\alpha} { 2}$<br>
6) $X_в = \dfrac{1}{n} \cdot \sum_{i=1}^{n}X_i$

In [4]:
x = [131, 125, 115, 122, 131, 115, 107, 99, 125, 111]
Xv = sum(x) / len(x)
print(f'Xв ={Xv: .4f}')

Xв = 118.1000


7) $s=\sqrt{\frac{1}{n-1}\cdot \sum_{i=1}^{n}(X_i-X_в)^2}$

In [5]:
s = math.sqrt(sum(map(lambda x: (x - Xv) ** 2, x)) / (10 - 1))

print(f's = {s: .4f}')

s =  10.5457


8) $K_{\frac{\alpha}{2}}=Z_{\frac{\alpha}{2}}$(из таблицы <font color = red>Стьюдента</font>)<br>
находим вероятность $p=1-\frac{\alpha}{2}$ (<font color = red>для $n-1$</font>)<br>
$Z_{\frac{\alpha}{2}} = K_{\frac{\alpha}{2}} = 2.262$


In [6]:
print(f'[{Xv - 2.262 * (s / math.sqrt(10)): .4f}; {Xv + 2.262 * (s / math.sqrt(10)): .4f}]')

[ 110.5566;  125.6434]


$[Xв - K \cdot Ст.ош.ср; Xв - K \cdot Ст.ош.ср]=[110.5566;  125.6434]$<br>

Вывод: $[ 110.5566;  125.6434]$ этот интервал покрывает истенное значение $X$ с доверительной вероятностью $95$%.

__3. Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.__

Дано:<br>
$\sigma = 5$<br>
$M(x) = a$<br>
$p = 0.95$<br>
$Xв. = 174.2$<br>
$n = 27$<br>

1) Уже были выполнен анализ ГС известно $\sigma$, значит критерий оценки $K$ является <font color = red>критерием Фишера</font> $Z$ из таблицы распределения <font color = red>Лапласа</font><br>
2) Надежность 0.95 => $p = 0.95$<br>
$\alpha = 1 - p => \alpha = 0.05$<br>
3) ГС распределена нормально => можем от $\alpha$ перейти к $\frac{\alpha} { 2}$<br>
$\alpha / 2 = 0.05 / 2 = 0.025$<br>
4) Из 1) => $Ст.ош.ср. = \sigma / \sqrt{n}$<br>
$[X_{в.} - Z_{\alpha/2} \cdot \dfrac{\sigma}{\sqrt{n}};X_{в.} + Z_{\alpha/2} \cdot \dfrac{\sigma}{\sqrt{n}}]$<br>
$Z(p=0.025)=Z(\alpha /2=0.025)=-1.96$<br>


In [9]:
a=math.sqrt(27)
print(f'[174.2 - 1.96(sigma/sqrt(27)); 174.2 + 1.96(sigma/sqrt(27))] = [{174.2 - 1.96 * (16. / a): .4f}; {174.2 + 1.96 * (16. / a): .4f}]')

[174.2 - 1.96(sigma/sqrt(27)); 174.2 + 1.96(sigma/sqrt(27))] = [ 168.1648;  180.2352]


Вывод: $[168.1648;  180.2352]$ - этот интервал с вероятностью 95% покрывает истенное $M(x)=a$ генеральной совокупности.